tool/modules/uds_fuzz.py

```diff
@@ -0,0 +1,473 @@
+from lib.can_actions import auto_blacklist
+from lib.common import list_to_hex_str, parse_int_dec_or_hex, str_to_int_list
+from lib.constants import ARBITRATION_ID_MAX, ARBITRATION_ID_MAX_EXTENDED
+from lib.constants import ARBITRATION_ID_MIN
+from lib.iso15765_2 import IsoTp
+from lib.iso14229_1 import Constants, Iso14229_1, NegativeResponseCodes, Services
+from sys import stdout, version_info
+import argparse
+import datetime
+import time
+
+
+# Python 2/3 compatibility
+if version_info[0] == 2:
+    range = xrange
+    input = raw_input
+
+NRC_NAMES = {
+    0x00: "POSITIVE_RESPONSE",
+    0x10: "GENERAL_REJECT",
+    0x11: "SERVICE_NOT_SUPPORTED",
+    0x12: "SUB_FUNCTION_NOT_SUPPORTED",
+    0x13: "INCORRECT_MESSAGE_LENGTH_OR_INVALID_FORMAT",
+    0x14: "RESPONSE_TOO_LONG",
+    0x21: "BUSY_REPEAT_REQUEST",
+    0x22: "CONDITIONS_NOT_CORRECT",
+    0x24: "REQUEST_SEQUENCE_ERROR",
+    0x25: "NO_RESPONSE_FROM_SUBNET_COMPONENT",
+    0x26: "FAILURE_PREVENTS_EXECUTION_OF_REQUESTED_ACTION",
+    0x31: "REQUEST_OUT_OF_RANGE",
+    0x33: "SECURITY_ACCESS_DENIED",
+    0x35: "INVALID_KEY",
+    0x36: "EXCEEDED_NUMBER_OF_ATTEMPTS",
+    0x37: "REQUIRED_TIME_DELAY_NOT_EXPIRED",
+    0x70: "UPLOAD_DOWNLOAD_NOT_ACCEPTED",
+    0x71: "TRANSFER_DATA_SUSPENDED",
+    0x72: "GENERAL_PROGRAMMING_FAILURE",
+    0x73: "WRONG_BLOCK_SEQUENCE_COUNTER",
+    0x78: "REQUEST_CORRECTLY_RECEIVED_RESPONSE_PENDING",
+    0x7E: "SUB_FUNCTION_NOT_SUPPORTED_IN_ACTIVE_SESSION",
+    0x7F: "SERVICE_NOT_SUPPORTED_IN_ACTIVE_SESSION",
+    0x81: "RPM_TOO_HIGH",
+    0x82: "RPM_TOO_LOW",
+    0x83: "ENGINE_IS_RUNNING",
+    0x84: "ENGINE_IS_NOT_RUNNING",
+    0x85: "ENGINE_RUN_TIME_TOO_LOW",
+    0x86: "TEMPERATURE_TOO_HIGH",
+    0x87: "TEMPERATURE_TOO_LOW",
+    0x88: "VEHICLE_SPEED_TOO_HIGH",
+    0x89: "VEHICLE_SPEED_TOO_LOW",
+    0x8A: "THROTTLE_PEDAL_TOO_HIGH",
+    0x8B: "THROTTLE_PEDAL_TOO_LOW",
+    0x8C: "TRANSMISSION_RANGE_NOT_IN_NEUTRAL",
+    0x8D: "TRANSMISSION_RANGE_NOT_IN_GEAR",
+    0x8F: "BRAKE_SWITCHES_NOT_CLOSED",
+    0x90: "SHIFT_LEVER_NOT_IN_PARK",
+    0x91: "TORQUE_CONVERTER_CLUTCH_LOCKED",
+    0x92: "VOLTAGE_TOO_HIGH",
+    0x93: "VOLTAGE_TOO_LOW"
+}
+
+# Number of seconds to wait between messages
+DELAY_SECSEED_RESET = 0.011
+DELAY_FUZZ_RESET = 3.901
+
+# Message data length limits
+MIN_DATA_LENGTH = 1
+MAX_DATA_LENGTH = 8
+# Max size of random seed if no seed is provided in arguments
+DEFAULT_SEED_MAX = 2 ** 16
+# Number of sub-lists to split message list into per round in 'replay' mode
+REPLAY_NUMBER_OF_SUB_LISTS = 5
+BYTE_MIN = 0x00
+BYTE_MAX = 0xFF
+
+# Duplicate testing from https://www.iditect.com/guide/python/python_howto_find_the_duplicates_in_a_list.html
+def find_duplicates(sequence):
+  first_seen = set()
+  first_seen_add = first_seen.add  
+  duplicates = set(i for i in sequence if i in first_seen or first_seen_add(i) )
+  return duplicates 
+
+def ecu_reset(arb_id_request, arb_id_response, reset_type, timeout):
```

Comment:
This should be imported from the UDS module, rather than copy-pasted into this module.